staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3068,6 +3068,50 @@ type ContainerStatus struct {
 	// +featureGate=SupplementalGroupsPolicy
 	// +optional
 	User *ContainerUser `json:"user,omitempty" protobuf:"bytes,13,opt,name=user,casttype=ContainerUser"`
+	// AllocatedResourcesStatus represents the status of various resources
+	// allocated for this Pod.
+	// +featureGate=ResourceHealthStatus
+	// +optional
+	AllocatedResourcesStatus ResourcesStatus `json:"allocatedResourcesStatus,omitempty" protobuf:"bytes,14,rep,name=allocatedResourcesStatus,casttype=ResourcesStatus,castkey=ResourceName"`
+}
+
+type ResourcesStatus map[ResourceName]ResourceStatus
+
+type ResourceStatus struct {
+	// map of unique Resource ID to its health.
+	// At a minimum, ResourceID must uniquely identify the Resource
+	// allocated to the Pod on the Node for the lifetime of a Pod.
+	// See ResourceID type for it's definition.
+	Resources map[ResourceID]ResourceHealth `json:"resources,omitempty" protobuf:"bytes,1,rep,name=resources,castkey=ResourceID"`
+
+	// allow to extend this struct in future with the overall health fields or things like Device Plugin version
+}
+
+// ResourceID is calculated based on the source of this resource health information.
+// For DevicePlugin:
+//
+//	deviceplugin:DeviceID, where DeviceID is from the Device structure of DevicePlugin's ListAndWatchResponse type: https://github.com/kubernetes/kubernetes/blob/eda1c780543a27c078450e2f17d674471e00f494/staging/src/k8s.io/kubelet/pkg/apis/deviceplugin/v1alpha/api.proto#L61-L73
+//
+// DevicePlugin ID is usually a constant for the lifetime of a Node and typically can be used to uniquely identify the device on the node.
+// For DRA:
+//
+//	dra:<driver name>/<pool name>/<device name>: such a device can be looked up in the information published by that DRA driver to learn more about it. It is designed to be globally unique in a cluster.
+type ResourceID string
+
+type ResourceHealth struct {
+	// Health of the resource.
+	// can be one of:
+	//  - Healthy: operates as normal
+	//  - Unhealthy: reported unhealthy. We consider this a temporary health issue
+	//               since we do not have a mechanism today to distinguish
+	//               temporary and permanent issues.
+	//  - Unknown: The status cannot be determined.
+	//             For example, Device Plugin got unregistered and hasn't been re-registered since.
+	//
+	// In future we may want to introduce the PermanentlyUnhealthy Status.
+	Health string `json:"health,omitempty" protobuf:"bytes,1,name=health"`
+
+	// TODO(SergeyKanzhelev, https://kep.k8s.io/4680): Add Conditions collection to track the history of resources health conditions updates
```

Comment:
Seeing this comment in a public facing API worries me. When are we going to implement conditions?

Reply:
next alpha when working on DRA changes. I can definitely move it out of public API. Since it is below any fields I thought it will not show anywhere in actual docs

Reply:
I can see that argument. I usually check the vendored directory when importing these APIs so I would see them if brought this API in.

Reply:
Moved a comment to the type definition.